### PR TITLE
Fix Multi-ASIC show specific resursive route by using common parsing function

### PR DESCRIPTION
**Pull request by gechiang**
#### What I did
On Multi-ASIC DUT when the route contains resursive routes in the nexthop information the show ip route handling parsing logic was missing the parsing logic for this case due to it was using its own parsing logic instead of sharing the previous parsing function that already handle the recursive next hop case.  This resulted to hitting an error during show handling similar to following:
```
admin@str2-n3164-acs-1:~$ show ip route -nasic4 -d all 193.11.208.0/25
Routing entry for 193.11.208.0/25
  Known via "bgp", distance 200, metric 0, best
  Last update 13:34:33 ago
Traceback (most recent call last):
  File "/usr/bin/show", line 12, in <module>
    sys.exit(cli())
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/show/main.py", line 1226, in route
    bgp_common.show_routes(args, namespace, display, verbose, "ip")
  File "/usr/lib/python2.7/dist-packages/show/bgp_common.py", line 423, in show_routes
    print_ip_routes(ns_route, filter_by_ip)
  File "/usr/lib/python2.7/dist-packages/show/bgp_common.py", line 159, in print_ip_routes
    str_2_print += ", via {}".format(info[i]['nexthops'][j]['interfaceName'])
KeyError: 'interfaceName'

```

#### How I did it
Changed the code to use the existing parsing function "get_nexthop_info_str()" that already handles recursive nexthop case.
This also improves code reuse.

#### How to verify it
After the fix in place perform similar show command and no longer see the error:

```
admin@str2-n3164-acs-1:~$ show ip route -nasic4 -d all 193.11.208.0/25
Routing entry for 193.11.208.0/25
  Known via "bgp", distance 200, metric 0, best
  Last update 13:34:33 ago
  * 10.0.0.1 (recursive)
  * 10.1.0.1, via PortChannel4004
  * 10.0.0.9 (recursive)
  * 10.1.0.5, via PortChannel4014

 admin@str2-n3164-acs-1:~$
```
Note this change only effects DUT that are Multi-ASIC based. no impact to non-Multi-ASIC platforms.

### CODE REVIEW

File: tests/mock_tables/asic2/ip_special_recursive_route.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    "193.11.208.0/25": [
```
Comment (judyjoseph) [inline tests/mock_tables/asic2/ip_special_recursive_route.json:2]: Confirming if this is the exact json output to be tested .. I see the following in the switch 
```
    "asic4": {
        "193.11.208.0/25": [
            {
                "destSelected": true, 
                "distance": 200, 
                "installed": true, 
                "internalFlags": 13, 
                "internalNextHopActiveNum": 2, 
                "internalNextHopNum": 2, 
                "internalStatus": 16, 
                "metric": 0, 
                "nexthops": [
                    {
                        "active": true, 
                        "afi": "ipv4", 
                        "fib": true, 
                        "flags": 3, 
                        "interfaceIndex": 128, 
                        "interfaceName": "PortChannel4014", 
                        "ip": "10.1.0.5"
                    }, 
                    {
                        "active": true, 
                        "afi": "ipv4", 
                        "flags": 5, 
                        "ip": "10.0.0.9", 
                        "recursive": true
                    }, 
                    {
                        "active": true, 
                        "afi": "ipv4", 
                        "fib": true, 
                        "flags": 3, 
                        "interfaceIndex": 126, 
                        "interfaceName": "PortChannel4004", 
                        "ip": "10.1.0.1"
                    }, 
                    {
                        "active": true, 
                        "afi": "ipv4", 
                        "flags": 5, 
                        "ip": "10.0.0.5", 
                        "recursive": true
                    }
                ], 
                "prefix": "193.11.208.0/25", 
                "protocol": "bgp", 
                "selected": true, 
                "table": 254, 
                "uptime": "20:17:37"
            }
        ]
    }, 
```

File: tests/mock_tables/asic1/ip_special_recursive_route.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (judyjoseph) [inline tests/mock_tables/asic1/ip_special_recursive_route.json:1]: This file can be removed ?

File: show/bgp_common.py [hunk 3]
```diff
@@ -152,13 +158,7 @@ def print_ip_routes(route_info, filter_by_ip):
...
                         print("  * directly connected, {}\n".format(info[i]['nexthops'][j]['interfaceName']))
                     else:
-                        if "ip" in info[i]['nexthops'][j]:
-                            str_2_print = "  * {}".format(info[i]['nexthops'][j]['ip'])
-
-                        if "active" in info[i]['nexthops'][j]:
-                            str_2_print += ", via {}".format(info[i]['nexthops'][j]['interfaceName'])
-                        else:
-                            str_2_print += " inactive"
+                        str_2_print = get_nexthop_info_str(info[i]['nexthops'][j], 1)
```
Comment (judyjoseph) [inline show/bgp_common.py:161]: Could use True/False instead of passing 1/0

Comment (gechiang) [reply show/bgp_common.py:161]: Agreed. Will make the change.  Please see my next commit update for this.

File: tests/mock_tables/asic1/ip_special_recursive_route.json (hunk 2 above)
Comment (gechiang) [reply tests/mock_tables/asic1/ip_special_recursive_route.json:1]: For the test command, it queries from all namespaces and it is needed to indicate there is no route from this particular namespace to see how it is handled.  So this is needed.

File: tests/mock_tables/asic2/ip_special_recursive_route.json (hunk 1 above)
Comment (gechiang) [reply tests/mock_tables/asic2/ip_special_recursive_route.json:2]: Yes.  I am not making it based on the DUT route but fabricating it new one.

**Review by judyjoseph — APPROVED** [review]
LGTM, please make sure the json output remains same with various FRR versions we have ( master : FRR_VERSION = 7.5.1 , 201911: FRR_VERSION = 7.2.1)

**Pull request merged**